tools/scripts/generic_variables.mk

```diff
@@ -29,7 +29,7 @@ PLATFORM_DRIVERS	?= $(NO-OS)/drivers/platform/$(PLATFORM)
 GIT_VERSION := $(shell git describe --all --long --dirty=-modified)
 GIT_VERSION := $(subst heads/,,$(GIT_VERSION))
 GIT_VERSION := $(subst -0-g,-,$(GIT_VERSION))
-CFLAGS += -DNO_OS_VERSION=$(GIT_VERSION)
+CFLAGS := -DNO_OS_VERSION=$(GIT_VERSION)
```

Comment:
Because i initialized CFLAGS in generic_variable make we must assign all the project related CFLAGS after generic_variable makefile include line like here [https://github.com/analogdevicesinc/no-OS/blob/e6a5052d2832529d50c51831a0a69c21d70a1c6f/projects/adrv9001/Makefile#L9 ](url) . Earlier  += variable initialization used for CFLAGS so When we do make all we will be calling submakes and because of this there is the repetition of CFLAGS. To remove CFLAGS repetition i used := variable initialization for CFLAGS.